tests/test_calc_barycentric_corr.py

```diff
@@ -0,0 +1,247 @@
+        # fmt: off
+        baryExpected = np.array([
+            #       dx,                      dxdx,                    dxdy,                    dy,                      dydx,                    dydy,
+            [     0.0000000000000000,      0.0000000000000000,      0.0000000000000000,      0.0000000000000000,      0.0000000000000000,      0.0000000000000000],
+            [    -1.1841515620267211,     -0.0000003401901527,      0.0000000009598210,     -2.9702163523510885,      0.0000000004351304,     -0.0000003379561716],
+            [    -1.9735671031906754,     -0.0000005674411201,      0.0000000015996900,     -4.9503275906180564,      0.0000000007252046,     -0.0000005637178156],
+            [  -100.9202381283850798,     -0.0000319903512675,      0.0000000817994955,   -253.2309265796297382,      0.0000000370963011,     -0.0000317998878157],
+            [  -102.1013035398333670,     -0.0000324007246175,      0.0000000827567619,   -256.1955780511078160,      0.0000000375305764,     -0.0000322080313739],
+            [  -102.8886591105074473,     -0.0000326747631053,      0.0000000833949229,   -258.1719728533235525,      0.0000000378200863,     -0.0000324805833438],
+            [  -201.5607485205129024,     -0.0000699606225079,      0.0000001633666874,   -505.9449406539421261,      0.0000000741141381,     -0.0000695800875374],
+            [  -202.7383401704241805,     -0.0000704409557553,      0.0000001643210668,   -508.9030324275871067,      0.0000000745474367,     -0.0000700581959558],
+            [  -203.5233773925634750,     -0.0000707616325463,      0.0000001649572917,   -510.8750476689951370,      0.0000000748362879,     -0.0000703773895892],
+            [  -301.8886731240853010,     -0.0001138896682213,      0.0000002446735039,   -758.0578046608508203,      0.0000001110405992,     -0.0001133195204643],
+        ])
+        # fmt: on
+        assert self._check_barycentric(baryCoeff, baryExpected)
+
+    def test_single_image(self):
+        """Verifies that the barycentric corrections for a single image are zeros."""
+        run_search = kbmod.run_search.run_search(self.input_parameters)
+        assert run_search is not None
+        # Load the PSF.
+        kb_interface = kbmod.analysis_utils.Interface()
+        default_psf = kbmod.search.psf(run_search.config["psf_val"])
+        full_file_path = f"{self.input_parameters['im_filepath']}/000000.fits"
+        header_info = kbmod.analysis_utils.ImageInfo()
+        header_info.populate_from_fits_file(full_file_path)
+        time_obj = header_info.get_epoch(none_if_unset=True)
+        time_stamp = time_obj.mjd
+        visit_times = [time_stamp]
+        img_info = kbmod.image_info.ImageInfoSet()
+        img_info.append(header_info)
+        img_info.set_times_mjd(np.array(visit_times))
+        baryCoeff = run_search._calc_barycentric_corr(img_info, 50.0)
+        assert baryCoeff is not None
+        assert baryCoeff.shape == (1, 6)
```

Comment:
These asserts can be changed into unittest style as above.

Reply:
yup. will do